pkg/registry/core/namespace/storage/storage.go

```diff
@@ -139,13 +139,21 @@ func (r *REST) Delete(ctx context.Context, name string, options *metav1.DeleteOp
 	}
 	if options.Preconditions.UID == nil {
 		options.Preconditions.UID = &namespace.UID
+		options.Preconditions.ResourceVersion = &namespace.ResourceVersion
```

Comment:
Does this need to be in a separate if block which checks `if options.Preconditions.ResourceVersion == nil`?

this question also applies to: https://github.com/kubernetes/kubernetes/blob/5e1beb3bff70e3284136bbf3f1ff40c9152dd78c/staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresourcedefinition/etcd.go#L87